File: backend/laravel-12.x/app/Models/Trade.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Support\Str;

class Trade extends Model
{
    use HasFactory;

    protected $fillable = [
        'trade_id',
        'user_id',
        'pair',
        'side',
        'type',
        'amount',
        'price',
        'total',
        'fee',
        'status',
        'executed_at',
        'metadata',
    ];

    protected $casts = [
        'amount' => 'decimal:8',
        'price' => 'decimal:8',
        'total' => 'decimal:8',
        'fee' => 'decimal:8',
        'executed_at' => 'datetime',
        'metadata' => 'array',
    ];

    protected static function boot()
    {
        parent::boot();
        
        static::creating(function ($trade) {
            if (empty($trade->trade_id)) {
                $trade->trade_id = 'TRD_' . Str::upper(Str::random(10));
            }
        });
    }

    public function user()
    {
        return $this->belongsTo(User::class);
    }

    public function cryptoPair()
    {
        return $this->belongsTo(CryptoPair::class, 'pair', 'symbol');
    }

    public function getFormattedAmountAttribute()
    {
        return number_format($this->amount, 8);
    }

    public function getFormattedPriceAttribute()
    {
        return number_format($this->price, 8);
    }

    public function getFormattedTotalAttribute()
    {
        return number_format($this->total, 2);
    }

    public function getSideColorAttribute()
    {
        return $this->side === 'buy' ? 'green' : 'red';
    }

    public function getStatusBadgeAttribute()
    {
        $colors = [
            'pending' => 'yellow',
            'completed' => 'green',
            'cancelled' => 'red',
            'failed' => 'red',
        ];

        return $colors[$this->status] ?? 'gray';
    }

    public function scopeCompleted($query)
    {
        return $query->where('status', 'completed');
    }

    public function scopeByPair($query, $pair)
    {
        return $query->where('pair', $pair);
    }

    public function scopeRecent($query)
    {
        return $query->orderBy('created_at', 'desc');
    }
}

File: backend/laravel-12.x/flutter_app/lib/models/trade.dart
class Trade {
  final int id;
  final String tradeId;
  final int userId;
  final String pair;
  final String side;
  final String type;
  final double amount;
  final double price;
  final double total;
  final double fee;
  final String status;
  final DateTime? executedAt;
  final DateTime createdAt;

  Trade({
    required this.id,
    required this.tradeId,
    required this.userId,
    required this.pair,
    required this.side,
    required this.type,
    required this.amount,
    required this.price,
    required this.total,
    required this.fee,
    required this.status,
    this.executedAt,
    required this.createdAt,
  });

  factory Trade.fromJson(Map<String, dynamic> json) {
    return Trade(
      id: json['id'],
      tradeId: json['trade_id'],
      userId: json['user_id'],
      pair: json['pair'],
      side: json['side'],
      type: json['type'],
      amount: double.parse(json['amount'].toString()),
      price: double.parse(json['price'].toString()),
      total: double.parse(json['total'].toString()),
      fee: double.parse(json['fee'].toString()),
      status: json['status'],
      executedAt: json['executed_at'] != null
          ? DateTime.parse(json['executed_at'])
          : null,
      createdAt: DateTime.parse(json['created_at']),
    );
  }

  String get formattedAmount => amount.toStringAsFixed(8);
  String get formattedPrice => '\$${price.toStringAsFixed(8)}';
  String get formattedTotal => '\$${total.toStringAsFixed(2)}';
  
  bool get isBuy => side.toLowerCase() == 'buy';
  bool get isCompleted => status.toLowerCase() == 'completed';
}
